git/git-clone.yaml

```diff
@@ -36,22 +36,71 @@ spec:
     description: The precise commit SHA that was fetched by this Task
   steps:
   - name: clone
-    image: gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init:latest
+    image: alpine/git:latest
     script: |
+      #!/usr/bin/env sh
+      # Most errors handling are handled by the set -eux
+      set -eux
       CHECKOUT_DIR="$(workspaces.output.path)/$(inputs.params.subdirectory)"
-      /ko-app/git-init \
-        -url "$(inputs.params.url)" \
-        -revision "$(inputs.params.revision)" \
-        -path "$CHECKOUT_DIR" \
-        -sslVerify "$(inputs.params.sslVerify)" \
-        -submodules "$(inputs.params.submodules)" \
-        -depth "$(inputs.params.depth)"
-      cd $CHECKOUT_DIR
-      RESULT_SHA="$(git rev-parse HEAD | tr -d '\n')"
-      EXIT_CODE="$?"
-      if [ "$EXIT_CODE" != 0 ]
-      then
-        exit $EXIT_CODE
-      fi
-      # Make sure we don't add a trailing newline to the result!
-      echo -n "$RESULT_SHA" > $(results.commit.path)
+      URL="$(inputs.params.url)"
+      REVISION="$(inputs.params.revision)"
+      SUBMODULES="$(inputs.params.submodules)"
+      SSLVERIFY="$(inputs.params.sslVerify)"
+      DEPTH="$(inputs.params.depth)"
+
+      # https://git.io/JvBCV
+      ensureHomeEnv() {
+          local realhomedir=$(getent passwd $(whoami)|cut -f6 -d:)
+          if [[ ! -d ${realhomedir} ]];then
+              mkdir -p ${realhomedir}
+          fi
+
+          if [[ ${realhomedir} == ${HOME} ]];then
+              return
+          fi
+
+          if [[ ! -e ${realhomedir}/.ssh ]];then
+              ln -svf ${HOME}/.ssh ${realhomedir}/.ssh
+          fi
+      }
+
+      # https://git.io/JvBCr
+      fetch() {
+          [[ ${REVISION} == "" ]] && REVISION=master
+          if [[ ${CHECKOUT_DIR} != "" ]];then
+              git init ${CHECKOUT_DIR}
+              cd ${CHECKOUT_DIR}
+          else
+              git init .
+          fi
+          trimmedURL=$(echo ${URL}|sed -e 's/^[ ]*//' -e 's/[ ]*$//')
```

Comment:
also just curious  ... wouldn't `trimmedURL=$(echo $URL)` work :D ?

Reply:
that doesn't trim the spaces: 

![image](https://user-images.githubusercontent.com/98980/77057148-ce659080-69d3-11ea-8ea9-24bcf9a6dbcb.png)

Reply:
![image](https://user-imagehubusercontent.com/3005132/77141007-27fab780-6ac7-11ea-9de5-eb1bd6108936.png)
🤔

Reply:
@chmouel are you using sh? or zsh?

Reply:
@chmouel you don't have to change the implementation, I was only curious ...